aries_vcx/src/handlers/out_of_band/receiver.rs

```diff
@@ -41,8 +41,8 @@ impl OutOfBandReceiver {
     pub async fn connection_exists<'a>(
         &self,
         pool_handle: PoolHandle,
-        connections: &'a Vec<&'a Connection>,
-    ) -> VcxResult<Option<&'a Connection>> {
+        connections: &'a Vec<&'a MediatedConnection>,
+    ) -> VcxResult<Option<&'a MediatedConnection>> {
```

Comment:
Wondering if we should delete this method and let upper layers track which connection "is who". 
As it's kind of code smell that we should now probably add similar support also for `Connection` instances, while the 2 are quite similar.